src/instructions/segmentation.rs

```diff
@@ -1,210 +1,278 @@
+            tmp = lateout(reg) _,
+            options(preserves_flags),
+        );
+
+        #[cfg(not(feature = "inline_asm"))]
+        crate::asm::x86_64_asm_set_cs(u64::from(sel.0));
+    }
 }
 
-/// Reload gs segment register.
+/// Stack Segment
 ///
-/// ## Safety
+/// Entirely unused in 64-bit mode, setting the segment register does nothing.
+#[derive(Debug)]
+pub struct SS;
+segment_impl!(SS, "ss", x86_64_asm_get_ss, x86_64_asm_load_ss);
+
+/// Data Segment
 ///
-/// This function is unsafe because the caller must ensure that `sel`
-/// is a valid gs segment descriptor.
-#[inline]
-pub unsafe fn load_gs(sel: SegmentSelector) {
-    #[cfg(feature = "inline_asm")]
-    asm!("mov gs, {0:x}", in(reg) sel.0, options(nostack, preserves_flags));
+/// Entirely unused in 64-bit mode, setting the segment register does nothing.
+#[derive(Debug)]
+pub struct DS;
+segment_impl!(DS, "ds", x86_64_asm_get_ds, x86_64_asm_load_ds);
 
-    #[cfg(not(feature = "inline_asm"))]
-    crate::asm::x86_64_asm_load_gs(sel.0);
-}
+/// ES Segment
+///
+/// Entirely unused in 64-bit mode, setting the segment register does nothing.
+#[derive(Debug)]
+pub struct ES;
+segment_impl!(ES, "es", x86_64_asm_get_es, x86_64_asm_load_es);
 
-/// Swap `KernelGsBase` MSR and `GsBase` MSR.
+/// FS Segment
 ///
-/// ## Safety
+/// Only base is used in 64-bit mode, see [`Segment64`]. This is often used in
+/// user-mode for Thread-Local Storage (TLS).
+#[derive(Debug)]
+pub struct FS;
+segment_impl!(FS, "fs", x86_64_asm_get_fs, x86_64_asm_load_fs);
+segment64_impl!(FS, "fs", FsBase, x86_64_asm_rdfsbase, x86_64_asm_wrfsbase);
+
+/// GS Segment
 ///
-/// This function is unsafe because the caller must ensure that the
-/// swap operation cannot lead to undefined behavior.
-#[inline]
-pub unsafe fn swap_gs() {
-    #[cfg(feature = "inline_asm")]
-    asm!("swapgs", options(nostack, preserves_flags));
+/// Only base is used in 64-bit mode, see [`Segment64`]. In kernel-mode, the GS
+/// base often points to a per-cpu kernel data structure.
+#[derive(Debug)]
+pub struct GS;
+segment_impl!(GS, "gs", x86_64_asm_get_gs, x86_64_asm_load_gs);
+segment64_impl!(GS, "gs", GsBase, x86_64_asm_rdgsbase, x86_64_asm_wrgsbase);
 
-    #[cfg(not(feature = "inline_asm"))]
-    crate::asm::x86_64_asm_swapgs();
+impl GS {
+    /// Swap `KernelGsBase` MSR and `GsBase` MSR.
+    ///
+    /// ## Safety
+    ///
+    /// This function is unsafe because the caller must ensure that the
+    /// swap operation cannot lead to undefined behavior.
+    pub unsafe fn swap() {
+        #[cfg(feature = "inline_asm")]
+        asm!("swapgs", options(nostack, preserves_flags));
+
+        #[cfg(not(feature = "inline_asm"))]
+        crate::asm::x86_64_asm_swapgs();
+    }
 }
 
-/// Returns the current value of the code segment register.
+/// Alias for [`CS::set_reg()`]
+#[deprecated(since = "0.14.4", note = "use `CS::set_reg()` instead")]
+#[inline]
+pub unsafe fn set_cs(sel: SegmentSelector) {
```

Comment:
Clippy complains that there is no "Safety" section in the docs of these unsafe functions. Given that they are only aliases now, I think we can just `#[allow]` that lint here (and for the other deprecated segment functions).

Reply:
Added in https://github.com/rust-osdev/x86_64/pull/258/commits/6164bd0232a3b0ff438d8c0a33c83b6ce56513bb